alvr/xtask/src/format.rs

```diff
@@ -0,0 +1,77 @@
+use alvr_filesystem as afs;
+use std::fs;
+use walkdir::{DirEntry, WalkDir};
+use xshell::{cmd, Shell};
+
+fn should_format(entry: &DirEntry) -> bool {
+    if entry.file_type().is_dir() {
+        match entry.path().file_name().unwrap().to_str().unwrap() {
+            "shared" | "include" => false,
+            _ => true,
+        }
+    } else {
+        entry
+            .path()
+            .extension()
+            .is_some_and(|ext| match ext.to_str().unwrap() {
+                "c" | "cpp" | "h" | "hpp" => true,
+                _ => false,
+            })
+            && entry.file_name().to_str().is_some_and(|name| match name {
+                "NvCodecUtils.h" | "NvEncoder.cpp" | "NvEncoder.h" | "NvEncoderD3D11.cpp" | "NvEncoderD3D11.h" => false,
+                _ => true,
+            })
+    }
```

Comment:
Since the expression is a bit complex, split into two booleans and then combine with &&